apps/compartmentConverter.cpp

```diff
@@ -162,9 +163,22 @@ int main( const int argc, char** argv )
     const brion::CompartmentCounts& counts = in.getCompartmentCounts();
     const brion::GIDSet& gids = in.getGIDs();
 
-    const lunchbox::URI outURI( vm.count( "output" ) == 1 ?
-                                vm[ "output" ].as< std::string >() :
-                                std::string( "out.h5" ));
+    lunchbox::URI outURI( vm.count( "output" ) == 1 ?
+                          vm[ "output" ].as< std::string >() :
+                          std::string( "out.h5" ));
+    if( outURI.getPath().empty( ))
+    {
+        try
+        {
+            outURI.setPath( boost::filesystem::canonical(
+                                inURI.getPath( )).generic_string( ));
+        }
+        catch( const boost::filesystem::filesystem_error& )
+        {
+            // For non-filebased reports, the canonical above will throw.
+            outURI.setPath( inURI.getPath( ));
```

Comment:
Yes thanks, I got confused between the path in the uri sense and the path in the filesystem sense (folder), especially because I saw that the default "out.h5" didn't have a "path"... that's still quite convoluted ;-)
And BTW, I know it's not directly what you were working on, but I think ( "output,o", po::value< std::string >(), "Output report URI" ) could be: ( "output,o", po::value< std::string >()->default_value("out.h5"), "Output report URI" )